services/src/main/java/org/exoplatform/addons/gamification/service/configuration/RuleServiceImpl.java

```diff
@@ -190,13 +190,11 @@ public RuleDTO createRule(RuleDTO ruleDTO, String username) throws IllegalAccess
     }
     ruleDTO.setCreatedBy(username);
     ruleDTO.setLastModifiedBy(username);
-    DomainDTO domainDTO = Utils.getDomainByTitle(ruleDTO.getArea());
```

Comment:
what about existing installations, should there be an upgrade plugin to change `Area` fiels of existing rules to use `programId` ? If yes, we should delete completely this field from database & DTO if the Model has evolved to keep the id only as unique way to link a rule to a program + we should add a foreign key on it for better evolutivity, coherence and maintainability.